etna/analysis/plotters.py

```diff
@@ -659,6 +659,55 @@ def plot_time_series_with_change_points(
         ax[i].tick_params("x", rotation=45)
 
 
+def get_residuals(forecast_df: pd.DataFrame, ts: "TSDataset") -> "TSDataset":
+    """Get residuals for further analysis.
+
+    Parameters
+    ----------
+    forecast_df:
+        forecasted dataframe with timeseries data
+    ts:
+        dataset of timeseries that was used for backtest
+
+    Returns
+    -------
+    new_ts:
+        TSDataset with residuals in forecasts
+
+    Raises
+    ------
+    ValueError:
+        if `forecast_tf` and `ts` don't match
+    ValueError:
+        if segments of `forecast_df` and `ts` don't match
+
+    Notes
+    -----
+    Transforms are taken as is from `ts`.
+    """
+    from etna.datasets import TSDataset
+
+    # make flatten dataframes for vectorized operations
+    true_df = TSDataset.to_flatten(ts[forecast_df.index, :, :]).sort_values(by=["segment", "timestamp"])
+    forecast_df = TSDataset.to_flatten(forecast_df).sort_values(by=["segment", "timestamp"])
+
+    # calculate residuals inside true_df
+    if len(true_df) != len(forecast_df):
+        raise ValueError("Lengths of `forecast_df` and `ts` don't match")
+    if not np.all(true_df["segment"] == forecast_df["segment"]):
+        raise ValueError("Segments of `forecast_df` and `ts` don't match")
+    true_df["target"] = true_df["target"] - forecast_df["target"]
+
+    # make TSDataset
+    new_df = TSDataset.to_dataset(true_df)
+    new_ts = TSDataset(df=new_df, freq=ts.freq)
+    new_ts.known_future = ts.known_future
+    new_ts._regressors = ts.regressors
+    new_ts.transforms = ts.transforms
```

Comment:
It seems transforms don't have any sense, do them?

Reply:
But if we have transforms in the original TSDataset it is strange to remove them.